### PR TITLE
Fix segfault: check for empty file names when constructing menus

### PR DESCRIPTION
**Pull request by veddox**
See #2079

### CODE REVIEW

File: shell_integration/nautilus/syncstate.py [hunk 1]
```diff
@@ -205,6 +205,7 @@ def check_registered_paths(self, filename):
     def get_file_items(self, window, files):
         # Show the menu extension to share a file or folder
 
+        if len(files) == 0: return [] #Appears to happen occasionally
```
Comment (misch7) [inline shell_integration/nautilus/syncstate.py:208]: Could you please add a space between `#` and `Appears` in the comment?
I'd also put the `return` in the next line (indented) and add a blank line after it.

File: src/gui/socketapi.cpp [hunk 2]
```diff
@@ -795,6 +795,10 @@ SyncJournalFileRecord SocketApi::FileData::journalRecord() const
 void SocketApi::command_GET_MENU_ITEMS(const QString &argument, OCC::SocketListener *listener)
 {
     listener->sendMessage(QString("GET_MENU_ITEMS:BEGIN"));
+	if (argument.isEmpty()) {
```
Comment (misch7) [inline src/gui/socketapi.cpp:798]: Please fix the indenting of the new `if` block and add blank lines around it :)

**Comment by veddox** [comment]
Thanks for the review :slightly_smiling_face: This should be it?

**Comment by veddox** [comment]
Great! I don't have write access, though - can you merge it?

File: shell_integration/nautilus/syncstate.py [hunk 3]
```diff
@@ -205,6 +205,8 @@ def check_registered_paths(self, filename):
     def get_file_items(self, window, files):
         # Show the menu extension to share a file or folder
 
+        if len(files) == 0:
+            return [] # Appears to happen occasionally
```
Comment (sbeyer) [inline shell_integration/nautilus/syncstate.py:209]: I hate to nitpick here but I would (like @misch7 mentioned) prefer a blank line after the `return` (please also make sure there is no whitespace on that line).

Moreover, I would prefer to have the comment after the `if` line, not the `return` line.

**Comment by veddox** [comment]
This PR may turn out to be redundant... See the discussion at #2079

**Comment by er-vin** [comment]
Yes, I'd say we check if the nautilus specific part fixes your double menu entry issue, otherwise better close this one and open a new one specifically for that other issue.

**Pull request closed without merging**